handlers/handlers.go

```diff
@@ -89,94 +80,108 @@ func FlorenceLoginHandlerPOST(ctx context.Context, usersFile string, privateKeyP
 		// Check both form and query parameters
 		username := req.FormValue("username")
 
+		if username == "" {
+			log.Error(ctx, "No username in request", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
 		redirect := req.FormValue("redirect")
 		if redirect == "" {
 			redirect = req.URL.Query().Get("redirect")
 		}
-		// Verify the user by email
-		user, err := utils.VerifyUser(ctx, usersFile, username)
+		// Get the user by email
+		user, err := store.GetUser(username)
 		if err != nil {
-			log.Error(ctx, "Invalid user", err)
+			log.Error(ctx, "Couldn't get user", err)
 			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
 
-		//userID := user.Username
 		cfg, _ := config.Get()
 
-		//generate the tokens
-		access_token := "Bearer " + generateJWT(*user, "access", *cfg, privateKeyPath)
-		id_token := generateJWT(*user, "id", *cfg, privateKeyPath)
+		// generate the tokens
+		accessToken, err := generateAccessTokenJWT(store, *user, cfg.AccessTokenValidityDuration)
+		if err != nil {
+			log.Error(ctx, "Failed to generate access token JWT", err)
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+
+		idToken, err := generateIDTokenJWT(store, *user, cfg.IDTokenValidityDuration)
+		if err != nil {
+			log.Error(ctx, "Failed to generate access token JWT", err)
+			w.WriteHeader(http.StatusInternalServerError)
+		}
 
-		refresh_token := "testrefreshtokennn" // Random opaque token string
+		refreshToken := "testrefreshtokennn" // Random opaque token string
 
 		// Store refresh token details in the in-memory map
-		models.RefreshTokenStore[refresh_token] = models.RefreshTokenInfo{
+		models.RefreshTokenStore[refreshToken] = models.RefreshTokenInfo{
 			Username:      username,
 			AuthTime:      time.Now(),
 			SessionExpiry: time.Now().Add(cfg.RefreshTokenValidityDuration), // Use your config for duration
 		}
 
-		//add to header
-		http.SetCookie(w, &http.Cookie{Name: "access_token", Value: access_token, Path: "/"})
-		http.SetCookie(w, &http.Cookie{Name: "id_token", Value: id_token, Path: "/"})
-		http.SetCookie(w, &http.Cookie{Name: "refresh_token", Value: refresh_token, Path: "/"})
+		// add to header
+		http.SetCookie(w, &http.Cookie{Name: models.AccessTokenCookie, Value: accessToken, Path: "/"})
+		http.SetCookie(w, &http.Cookie{Name: models.IDTokenCookie, Value: idToken, Path: "/"})
+		http.SetCookie(w, &http.Cookie{Name: models.RefreshTokenCookie, Value: refreshToken, Path: "/"})
```

Comment:
Minor, while you are here can we update `Path` for refresh to `/tokens/self` and add HttpOnly flags. Applies to other SetCookie calls.